src/map.cpp

```diff
@@ -4023,7 +4023,8 @@ bool map::add_item_or_charges(const tripoint &p, item new_item, int overflow_rad
             }
         }
         if( i_at( p_it ).size() < MAX_ITEM_IN_SQUARE ) {
-            add_item( p_it, new_item );
+            if( !( new_item.has_flag("IRREMOVABLE") && new_item.is_gunmod() ) )
```

Comment:
The problem is, that guns for some reason gets the flags of it's mods. so unless I check for it, it would remove any gun with a mod with that tag.